data-acquisition/data-synthesizer/src/template_manager/template_manager.py

```diff
@@ -0,0 +1,146 @@
+"""
+Module to manage the addition, deletion and modification of foreground
+and background images inside the img_templates folder. The configurations
+will be stored in the files in config folder.
+"""
+
+import os
+import json
+
+
+class TemplateManager:
+    def __init__(self, config_path):
+        self.config_path = config_path
+
+    def add_foreground_category(self, category):
+        """
+        Adding new foreground category.
+        """
+        config_path = self.config_path + "foregrounds.json"
+        with open(config_path, "r") as conf_file:
+            config = json.load(conf_file)
+        if category not in config:
+            config[category] = {}
+        else:
+            raise Exception((
+                "Category {category} present in foreground configuration."
+            ).format(category=category))
+
+        with open(config_path, "w") as conf_file:
+            json.dump(config, conf_file, indent=4)
+        print("New foreground category {} added.".format(category))
+
+    def del_foreground_category(self, category):
+        """
+        Removing foreground category.
+        """
+        config_path = self.config_path + "foregrounds.json"
+        with open(config_path, "r") as conf_file:
+            config = json.load(conf_file)
+        if category in config:
+            config.pop(category)
+        else:
+            raise Exception((
+                "Category {} not present in foreground configuration."
+            ).format(category))
+
+        with open(config_path, "w") as conf_file:
+            json.dump(config, conf_file, indent=4)
+        print("Removed category {} and all associated objects.".format(
+            category))
+
+    def _add_foreground_template(self, label, config, args):
+
+        category = args["category"]
+        if category in config:
+            if label not in config[category]:
```

Comment:
We should invert this logic, at both levels:

```python
if category not in config:
    raise Exception((
        "Category {category} not found in foreground configuration."
        " If you are adding a new category object please use "
        "the function `add_foreground_category`").format(category=category))

if label in config[category]:
    raise Exception((
        "Foreground object already exists in {category}."
        " If you want to modify an existing configuration use "
        "the function `modify_template`").format(category=category))

config[category][label] = {
    "height": args["height"],
    "width": args["width"]
}
```

Reasons:
1. Convention is to throw exceptions early on
2. It makes the error conditions more explicit (as opposed to occurring in `else` sections)

Reply:
Further, the correct error to throw is [`RuntimeError`](https://docs.python.org/3/library/exceptions.html#RuntimeError), which is a concrete error. [`Exception`](https://docs.python.org/3/library/exceptions.html#Exception) is a [base class](https://docs.python.org/3/library/exceptions.html#BaseException) and should not be thrown as is.